internal/app/coroutines/completePromise.go

```diff
@@ -165,8 +165,9 @@ func CompletePromise(metadata *metadata.Metadata, req *t_api.Request, res func(*
 			} else {
 				status := t_api.ForbiddenStatus(p.State)
 				strict := req.CompletePromise.Strict && p.State != req.CompletePromise.State
+				timeout := !req.CompletePromise.Strict && req.CompletePromise.State.In(promise.Timedout)
```

Comment:
```suggestion
				timeout := !req.CompletePromise.Strict && req.CompletePromise.State == promise.Timedout
```

Since we're only checking for one state, we can do a direct comparison